mockwebserver/src/main/java/okhttp3/mockwebserver/HeldCertificate.java

```diff
@@ -0,0 +1,332 @@
+/*
+ * Copyright (C) 2016 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package okhttp3.mockwebserver;
```

Comment:
one interesting question: should this go in okhttp3 instead? Currently this class has a bouncycastle dependency, but it’s plausible that we can hack these things in manually.

Reply:
The documentation seems like it's trying to convince me that it should stay here. Granted I don't know a ton about the subject.